[23.11] nextcloud28: 28.0.5 -> 28.0.6, nextcloud29: 29.0.0 -> 29.0.1

teutat3s opened the pull request:
## Description of changes
Manual backport of https://github.com/NixOS/nixpkgs/pull/290507 https://github.com/NixOS/nixpkgs/pull/298348 #314096

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

dotlambda committed:
nextcloud-notify_push.test_client: add meta

(cherry picked from commit 16e379fa03c0f4b9d9d9f39920cf08a0b78e2ef0)

r-ryantm committed:
nextcloud-notify_push: 0.6.9 -> 0.6.10

(cherry picked from commit 103d4f8a494025a585cb38ab018b44af98851496)

dotlambda committed:
nextcloud28: 28.0.5 -> 28.0.6

Changelog: https://nextcloud.com/changelog/#28-0-6
(cherry picked from commit 84c8e8af002131881ec7bf62d13b24a189a9b860)

dotlambda committed:
nextcloud29: 29.0.0 -> 29.0.1

Changelog: https://nextcloud.com/changelog/#29-0-1
(cherry picked from commit ff64ac2e176d2a71d5b07c1ea88512b73fc38e43)

dotlambda committed:
nextcloud-notify_push: 0.6.10 -> 0.6.11

Diff: https://github.com/nextcloud/notify_push/compare/v0.6.10...v0.6.11

Changelog: https://github.com/nextcloud/notify_push/releases/tag/v0.6.11
(cherry picked from commit dd7a48b54fc63a165cde8300f232f375f50431dc)

dotlambda committed:
nextcloudPackages: update

(cherry picked from commit ea23d5ac3d277f8986135aed32feac9fc477fc03)

risicle commented:
Assume the discrepancy in the CI is just because https://github.com/NixOS/nixpkgs/commit/3823b98cb5e610fcf16a1ed18addd935c7bff5fe didn't get backported

teutat3s commented:
> Assume the discrepancy in the check-cherry-pick CI is just because [3823b98](https://github.com/NixOS/nixpkgs/commit/3823b98cb5e610fcf16a1ed18addd935c7bff5fe) didn't get backported

A `git blame` points at 3823b98cb5e610fcf16a1ed18addd935c7bff5fe which was never backported. There was a conflict I resolved, that's what the cherry-pick discrepancy shows.

teutat3s commented:
`notify_push` build currently fails:
```
error: package `notify_push v0.1.0 (/build/source)` cannot be built because it requires rustc 1.74.0 or newer, while the currently active rustc version is 1.73.0
```

teutat3s committed:
nextcloud-notify_push: build with rustPackages_1_76

Minimum rustc version is 1.74 since notify_push version 0.6.10
https://github.com/nextcloud/notify_push/commit/9a8d0be4f004d54a4761aa056ab1b4579f388fc0
